src/V3Width.cpp

```diff
@@ -2343,6 +2343,13 @@ class WidthVisitor final : public VNVisitor {
         // if (debug() >= 9) nodep->dumpTree("-  VRout: ");
         if (nodep->access().isWriteOrRW() && nodep->varp()->direction() == VDirection::CONSTREF) {
             nodep->v3error("Assigning to const ref variable: " << nodep->prettyNameQ());
+        } else if (!nodep->varp()->isRand() && !nodep->varp()->isForced()
```

Comment:
I can take it out, I don't believe we have a test for it anyway. I had shamelessly stolen it from this pre-linking section:
https://github.com/verilator/verilator/blob/master/src/V3LinkLValue.cpp#L48

Where it would return if isRand(), and thus not reach the later `nodep->v3warn(ASSIGNIN, "Assigning to input/const variable: " ... ) `

Reply:
I can't craft a valid example that reaches this part of V3Width.cpp AND tolerates an AstVar with isRand()=true, so I removed it.